WordPress/Classes/Stores/StatsWidgetsStore.swift

```diff
@@ -303,6 +304,38 @@ private extension StatsWidgetsStore {
             self?.initializeStatsWidgetsIfNeeded()
         }
     }
+
+    /// Observes applicationLaunchCompleted notification and runs migration.
+    func observeApplicationLaunched() {
+        guard #available(iOS 14.0, *) else {
+            return
+        }
+
+        NotificationCenter.default.addObserver(forName: NSNotification.Name.applicationLaunchCompleted,
+                                               object: nil,
+                                               queue: nil) { [weak self] _ in
+            self?.handleJetpackWidgetsMigration()
+        }
+    }
+}
+
+private extension StatsWidgetsStore {
+
+    /// Handles migration to a Jetpack app version that started supporting Stats widgets.
+    /// The required flags in shared UserDefaults are set and widgets are initialized.
+    func handleJetpackWidgetsMigration() {
+        // If user is logged in but defaultSiteIdKey is not set
+        guard let account = try? WPAccount.lookupDefaultWordPressComAccount(in: blogService.managedObjectContext),
+              let siteId = account.defaultBlog?.dotComID,
+              let userDefaults = UserDefaults(suiteName: WPAppGroupName),
+              userDefaults.value(forKey: AppConfiguration.Widget.Stats.userDefaultsSiteIdKey) == nil else {
+            return
+        }
+
+        userDefaults.setValue(AccountHelper.isLoggedIn, forKey: AppConfiguration.Widget.Stats.userDefaultsLoggedInKey)
```

Comment:
Note: Check if we could check existence of access token with a static key

Reply:
>  I thought the risk was that the user adds widgets before launching Jetpack

I don't see how can we avoid this risk, even if we make changes on what Widgets depends on. For those users who are updating the app from an older version that didn't support widgets, we need to call `AccountService setupAppExtensionsWithDefaultAccount` at least once, so access token would be set in the shared keychain with the expected key. Even though the key would be static, they would still differ from `WordPress` and would need to be set at least once.